core/templates/dev/head/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -90,6 +90,8 @@ angular.module('oppia').directive('ckEditor4Rte', [
           'bulletedList', 'pre', 'indent', 'outdent'];
 
         // Add external plugins.
+        // prevents ckeditor from filtering content.
+        CKEDITOR.config.allowedContent = true;
```

Comment:
I am a bit confused about this. Shouldn't this already be accounted for by the extraAllowedContentRules on line 115, which makes use of the componentRule in line 61?

Reply:
Yeah, the Ckeditor behaviour is pretty unpredictable, Even after adding this line ckeditor still was removing span tags.
I too am confused in this aspect.